.github/workflows/release.yml

```
@@ -0,0 +1,96 @@
+name: release
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
```

```suggestion
      - master
```

But in general I would be opposed to doing a release only on a `workflow_dispatch` for extra security